invoke-ai/InvokeAI
# Feat/auto assign board on click

## What type of PR is this? (check all applicable)



## Have you discussed this change with the InvokeAI team?

      
## Have you updated all relevant documentation?


## Description
Add feature auto assign board on click. There is a new setting in the board menu. It is checked by default. When checked the auto assign board is changed whenever selecting a board. The auto assign board will not change during image generation. If the new feature is disabled in the settings the original method of assigning a board returns.

## Related Tickets & Documents

- Related Issue #4018
- Closes #4018

## QA Instructions, Screenshots, Recordings

Open the board menu by clicking the dropdown with the selected board. Click the settings icon. Note that the new option Auto-Assign Board on Click is selected. When this option is enabled the auto assign board switches by changing the active board. When Auto-Assign Board on Click is not selected you get a dropdown to assign the active board. The board assignment should not change during image generation.

![Screenshot 2023-07-28 103331](https://github.com/invoke-ai/InvokeAI/assets/13532991/e05615e7-cea8-40a4-9c0d-a36d79e7798a)
![Screenshot 2023-07-28 103301](https://github.com/invoke-ai/InvokeAI/assets/13532991/d116567d-1631-498e-b56d-8b9e65e19de9)


## Added/updated tests?

- [x] No : I will add tests when I know which ones to use

## [optional] Are there any post deployment tasks we need to perform?
No
